File: src/main/java/lightninglobby/lightninglobby/Serverjoin.java
package lightninglobby.lightninglobby;

import org.bukkit.Color;
import org.bukkit.GameMode;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.PlayerInventory;
import org.bukkit.inventory.meta.ItemMeta;
import org.jetbrains.annotations.NotNull;

public class Serverjoin implements Listener {
    @EventHandler
    public void PlayerJoinEvent(PlayerJoinEvent event) {
        Player player = event.getPlayer();
        PlayerInventory inv = player.getInventory();
        inv.clear();
        if (player.getLocation().getWorld().getName().equals("Minigames_Lobby")) {
            ItemStack compass = new ItemStack(Material.COMPASS, 1);
            ItemMeta commeta = compass.getItemMeta();
            commeta.setDisplayName("Games");
            compass.setItemMeta(commeta);
            player.setCanPickupItems(false);
            player.setHealth(20);
            player.setInvisible(false);
            player.setGameMode(GameMode.ADVENTURE);
            inv.setItem(4, compass);

        }
    }
}

File: src/main/kotlin/lightninglobby.lightninglobby/Serverjoin.kt
package lightninglobby.lightninglobby

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.GameMode
import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.inventory.ItemStack

class Serverjoin : Listener {
    @EventHandler
    fun PlayerJoinEvent(event: PlayerJoinEvent) {
        val player = event.player
        val inv = player.inventory
        inv.clear()
        if (player.location.world.name == "Minigames_Lobby") {
            val compass = ItemStack(Material.COMPASS, 1)
            val commeta = compass.itemMeta
            commeta.displayName(Component.text("Games").color(NamedTextColor.DARK_BLUE))
            compass.itemMeta = commeta
            player.canPickupItems = false
            player.health = 20.0
            player.isInvisible = false
            player.gameMode = GameMode.ADVENTURE
            inv.setItem(4, compass)
        }
    }
}